core/DataAccess/LogQueryBuilder/JoinGenerator.php

```diff
@@ -234,7 +234,13 @@ public function sortTablesForJoin($tA, $tB)
             if (isset($tA['tableAlias']) && isset($tB['joinOn']) && strpos($tB['joinOn'], $tA['tableAlias']) !== false) {
                 return -1;
             }
-            return 0;
+
+            $tA = $tA['table'];
+            $tB = $tB['table'];
+
+            if ($tA === $tB) {
+                return 1; // if both join same table keep order
+            }
```

Comment:
Maybe even better would be to look up the index in the array that is being sorted and simply always keep the order but this would mean we need to use a closure when sorting with `function ($tA, $tB) use ($tablesToBeSorted) `. Could do it but would mean a little refactoring